### PR TITLE
Add parameter snapshotTemplate

### PR DESCRIPTION
**Pull request by JamieMagee**
Allow customisation of the SNAPSHOT timestamp format through use of a template

### CODE REVIEW

**Comment by tcurdt** [comment]
Thanks for the PR. Looks OK to me except for the following:
- the actual parsing and replacement should be done in a separate function. maybe in even as part of the utils.
- adjusting the docs would be good

File: src/main/java/org/vafer/jdeb/utils/Utils.java [hunk 1]
```diff
@@ -235,21 +235,36 @@ public static String replaceVariables( final VariableResolver pResolver, final S
...
             if (apply) {
                 final String envValue = System.getenv(envName);
-                SimpleDateFormat dateFormat = new SimpleDateFormat("yyyyMMddHHmmss");
-                dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
-                final String snapshot = (envValue != null && envValue.length() > 0)
-                        ? envValue
-                        : dateFormat.format(timestamp);
-                version += snapshot;
+                if(template != null && template.length() > 0) {
+                    int startBracket = template.indexOf('[');
```
Comment (tcurdt) [inline src/main/java/org/vafer/jdeb/utils/Utils.java]: That's the code I was referring to. The parsing and replacement should be separated out.

Comment (JamieMagee) [reply src/main/java/org/vafer/jdeb/utils/Utils.java]: Sure. Just the template code, or do you want the whole if/elseif/else extracted into a separate function?

**Comment by JamieMagee** [comment]
Thanks for the (very) quick feedback. If I am able to fix the issues soon, will this be included in the 1.7 release?

**Comment by tcurdt** [comment]
Yes, I think we can sneak that into 1.7.

File: src/main/java/org/vafer/jdeb/utils/Utils.java (hunk 1 above)
Comment (tcurdt) [reply src/main/java/org/vafer/jdeb/utils/Utils.java]: I was thinking the parsing and replacing. Similar to the variable resolving. That makes it more easily testable.

**Comment by tcurdt** [comment]
Looking good. Thanks!

**Pull request merged**